nginx-proxy/default.conf

```diff
@@ -0,0 +1,103 @@
+log_format log_json escape=json '{'
+  '"body_bytes_sent": $body_bytes_sent,'
+  '"content_length": "$content_length",'
+  '"content_type": "$content_type",'
+  '"gzip_ratio": "$gzip_ratio",'
+  '"host": "$host",'
+  '"http_cf_ray": "$http_cf_ray",'
+  '"http_dnt": "$http_dnt",'
+  '"http_referer": "$http_referer",'
+  '"http_user_agent": "$http_user_agent",'
+  '"http_x_forwarded_for": "$http_x_forwarded_for",'
+  '"remote_addr": "$remote_addr",'
+  '"remote_user": "$remote_user",'
+  '"request_completion": "$request_completion",'
+  '"request_method": "$request_method",'
+  '"request_time": "$request_time",'
+  '"request_uri": "$request_uri",'
+  '"status": $status,'
+  '"time_local": "$time_local",'
+  '"type": "nginx",'
+  '"upstream_cache_status": "$upstream_cache_status"'
+'}';
+
+upstream backend {
+  server ${BACKEND_HOST};
+}
+
+server {
+  listen ${PORT};
+
+  proxy_buffers 16 16k;
+  proxy_buffer_size 16k;
+
+  location / {
+    proxy_pass http://backend;
+  }
+
+  access_log /dev/stdout log_json;
+
+  proxy_intercept_errors on;
+  proxy_ssl_server_name on;
+
+  proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+  proxy_set_header X-Real-IP  $remote_addr;
+  proxy_set_header Host $http_host;
+  proxy_set_header If-Modified-Since $http_if_modified_since;
```

Comment:
I understand we need to overwrite the `Authorization` header and `Host` header for each error page. Why are we setting these particular 4 headers as well?

Reply:
I copied this from our [current NGINX config](https://github.com/futurelearn/chef/blob/master/site-cookbooks/app/templates/default/futurelearn_nginx_conf.erb#L141-L144).

The main thing is `X-Forwarded-For` - I wanted to ensure we continued forwarding the remote user IP.